libnetwork/ipams/defaultipam/structures.go

```diff
@@ -59,13 +77,28 @@ func PoolIDFromString(str string) (pID PoolID, err error) {
 	return pID, nil
 }
 
+func parsePoolIDV2(str string) (pID PoolID, err error) {
+	data := strings.TrimPrefix(str, poolIDV2Prefix)
+
+	if err := json.Unmarshal([]byte(data), &pID); err != nil {
+		return PoolID{}, err
```

Comment:
This should probably get the `types.InvalidParameterErrorf()` treatment too? (Although they might all be internal errors really, as this isn't user input?)

Reply:
I fixed that in the original commit and added a new one to convert all `InvalidParameter` into `InternalError`s.